### ISSUE TITLE
`source` command turns off auto-completion and syntax highliting

### ISSUE THREAD

**Issue by kubouch** [nushell/nushell#3546]
**Describe the bug**
For some reason, auto-completion doesn't work after typing "source".

**To Reproduce**
1. `touch file_that_exists.txt<Enter>`
2. `source file_that_ex` (no syntax highliting)
3. hit `<Tab>`, nothing happens

**Expected behavior**
Auto-completion and syntax highliting should work the same as in other commands.

**Screenshots**

**Configuration (please complete the following information):**

Run `version | pivot` and paste the output to show OS, features, etc.

```
> version | pivot
  #         Column0                                                      Column1
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   0   version              0.32.0
   1   branch               main
   2   short_commit         290c712c
   3   commit_hash          290c712cde3d84873cd7b482ae9356f28fad6b35
   4   commit_date          2021-06-02 10:49:14
   5   build_os             linux-x86_64
   6   rust_version         rustc 1.52.0 (88f19c6da 2021-05-03)
   7   rust_channel         stable (default)
   8   cargo_version        cargo 1.52.0 (69767412a 2021-04-21)
   9   pkg_version          0.32.0
  10   build_time           2021-06-02 16:11:00
  11   build_rust_channel   release
  12   features             clipboard-cli, ctrlc, default, directories, dirs, ptree, rustyline, term, trash, uuid, which, zip

**Comment by ahkrr** [comment]
This behavior exists, because `source` has special casing in the `parser`.
This interferes with highlighting and completion. 
```rust
    // crates/nu-parser/src/parse.rs +1770
    } else if lite_cmd.parts[0].item == "source" {
        if lite_cmd.parts.len() != 2 {
            return (
                None,
                Some(ParseError::argument_error(
                    lite_cmd.parts[0].clone(),
                    ArgumentError::MissingMandatoryPositional("a path for sourcing".into()),
                )),
            );
        }
        if lite_cmd.parts[1].item.starts_with('$') {
            return (
                None,
                Some(ParseError::mismatch(
                    "a filepath constant",
                    lite_cmd.parts[1].clone(),
                )),
            );
        }
        if let Ok(contents) =
            std::fs::read_to_string(expand_path(&lite_cmd.parts[1].item).into_owned())
        {
            let _ = parse(&contents, 0, scope);
        } else {
            return (
                None,
                Some(ParseError::mismatch(
                    "a filepath to a source file",
                    lite_cmd.parts[1].clone(),
                )),
            );
        }
    }
```

There may be plans for the `source` command in the future, so this might change.
This PR #3148 that was trying to extend `source` (and if I remember correctly, also fix completion and highlighting) wasn't merged because of future design considerations for sourcing behavior.

**Comment by jonathandturner** [comment]
Should be fixed with #3589
